sotodlib/site_pipeline/update_obsdb.py

```diff
@@ -0,0 +1,160 @@
+        tube_slot: str
+        type: str
+
+"""
+from sotodlib.core.metadata import ObsDb
+from sotodlib.core import Context 
+from sotodlib.site_pipeline.check_book import main as checkbook
+import os
+import glob
+import yaml
+import numpy as np
+import time
+import argparse
+
+
+def check_meta_type(bookpath):
+    metapath = os.path.join(bookpath, "M_index.yaml")
+    meta = yaml.safe_load(open(metapath, "rb"))
+    if meta is None:
+        return "empty"
+    elif "type" not in meta:
+        return "notype"
+    else:
+        return meta["type"]
+
+def update_obsdb(base_dir, 
+                 config="config.yaml", 
+                 recency=2., 
+                 verbosity=2, 
+                 booktype="both"):
+    """
+        Create or update an obsdb for observation or operations data.
+    Argument
+    ----------
+    base_dir : str
+        The base directory in which to look for books
+    Parameters
+    ----------
+    config : str
+        Path to config file
+    recency : float
+        How far back in time to look for databases, in days. (default: 2.)
+    booktype : str
+        Look for observations or operations data or both (default: both)
+    """
+    bookcart = []
+    bookcartobsdb = ObsDb()
+
+    if booktype=="both":
+        accept_type = ["obs", "oper"]
+    else:
+        accept_type = [booktype]
+
+    configs = yaml.safe_load(open(config, "r"))
+    if "obsdb" in configs:
+        if os.path.isfile(configs["obsdb"]):
+            bookcartobsdb = ObsDb.from_file(configs["obsdb"])
+    if "cols" in configs:
+        col_list = []
+        for col, typ in configs["cols"].items():
+            col_list.append(col+" "+typ)
+        bookcartobsdb.add_obs_columns(col_list)
+
+    #How far back we should look
+    tnow = time.time()
+    tback = tnow - recency*86400
+    #Find folders that are book-like and recent
+    for dirpath,_, _ in os.walk(base_dir):
+        last_mod = max(os.path.getmtime(root) for root,_,_ in os.walk(dirpath))
+        if last_mod<tback:#Ignore older directories
+            continue
+        if os.path.exists(os.path.join(dirpath, "M_index.yaml")):
+            #Looks like a context file
+            bookcart.append(dirpath)
+    #Check the books for the observations we want
+
+    for bookpath in bookcart:
+        if check_meta_type(bookpath) in accept_type:
+            """ obsfiledb creation
+            The extra files give us a bit of trouble. We'll assume that index,
+            book, and bookbinder_log are always here, and edit our config file
+            to cover the rest as extra files.
+            """
+            book_file_list = os.listdir(bookpath)
+            book_file_list.remove("M_index.yaml")
+            book_file_list.remove("M_book.yaml")
+            book_file_list.remove("Z_bookbinder_log.txt")
+            book_file_list = [file_name for file_name in book_file_list if not file_name.endswith(".g3")]
+            if len(book_file_list)>0:
```

Comment:
I think I understand why this is here. It's because the `oper` books have extra files. @mhasself  I'd rather update `check_book` to be a bit nicer about the presence of extra files in the books. Can we do that?

Reply:
The problem actually arose when one `obs` book had an extra file called "frame_splits.txt". But you're right that check_book is unable to ignore extra files like this.